### PR TITLE
Add code formatting button

### PR DESCRIPTION
**Pull request by jjroley**

### CODE REVIEW

**Review by EthanLuisMcDonough — CHANGES_REQUESTED** [review]
Thanks for the submission!

**Comment by jjroley** [comment]
Oh, nice! I wasn't aware of the data in `ScratchpadUI`

File: src/editor-settings.ts [hunk 1]
```diff
@@ -332,6 +377,40 @@ function addEditorSettings (toggleButton: HTMLElement, editor: HTMLElement) {
...
+    const formatCodeButton = document.createElement('button')
+    formatCodeButton.textContent = "Format Code"
+    formatCodeButton.addEventListener('click', () => {
+      if(SUPPORTED_TYPES.includes(programType)) {
+        formatCode(programType)
+      }
+    })
+    table.appendChild(formatCodeButton)
+
+    const { type } = getParams('/computer-programming/new/:type')
```
[the remark this answers is not in the document]
Comment (jjroley) [reply src/editor-settings.ts]: Oh, nice! I wasn't aware of the `ScratchpadUI` object

**Review by EthanLuisMcDonough — APPROVED** [review]
All right, looks good to me.  Thanks for contributing!

**Pull request merged**